torch_np/_wrapper.py

```diff
@@ -497,10 +492,35 @@ def ravel_multi_index(multi_index, dims, mode='raise', order='C'):
     return sum(idx*dim for idx, dim in zip(multi_index, dims))
 
 
-# YYY : pattern: array_like input, tuple of arrays as output; cf broadcast_arrays
 def nonzero(a):
-    a = asarray(a).get()
-    return tuple(asarray(_) for _ in a.nonzero(as_tuple=True))
+    arr = asarray(a)
+    return arr.nonzero()
+
+
+def flatnonzero(a):
+    arr = asarray(a)
+    return nonzero(arr.ravel())[0]
+
+
+def argwhere(a):
+    arr = asarray(a)
+    if arr.ndim == 0:
+        a = atleast_1d(a)
+        # then remove the added dimension
+        return argwhere(a)[:, :0]
```

Comment:
This cannot be right. Did you mean
```suggestion
        return argwhere(a)[:, 0]
```
?